Create Mock Data (#5, #7, #8)

zhandavidz opened the pull request:
- documented and determined data schema for IndexedDB on the client-side
	- in `docs/data.md`
- generated the data using [mockaroo.com](https://mockaroo.com/)
- added the data as `.json` files
	- in directory `frontend/src/api/mock-data`

Will complete issue #5 
- Closes #7 
- Closes #8

zhandavidz committed:
feat: Document data storage format for IndexedDB

- added coum

zhandavidz committed:
feat: Generate & add mock data as `.json`

- Generated mock data on mockaroo
- Added mock data for listings, messages, reviews, and users into repo as `JSON` files
- Updated `data.md` with links to mockaroo schemas

Resolves #5 [and resolves subtask #8]

zhandavidz committed:
fix: Correct mistakes in data documentation

- minor fixes in `data.md`

Resolves #5 [subissues #7, #8]

lrdiaz reviewed (commented):
Looks good to me! I know that the timestamp data type is used for the "sent_at" field in messages.json and "created_at" field in reviews.json, but I'm not sure if there were supposed to be any hyphens or colons there.

zhandavidz commented:
Thanks! Yeah, the timestamp data type is actually just the datetime stored as epoch (# of seconds since 1/1/1970 00:00:00 UT), which is why it shows up as just a string of a large number. I think this is the easiest way of storing a datetime that means the same thing across potentially multiple database schemas, but this wasn't very clearly documented, so I will add that, thanks for catching that!

zhandavidz committed:
fix: Better specify the datatype of timestamp

- specified that timestamps are stored as a string representation of the epoch in seconds

Resolves #5 [subissue #7]

lhlee-umass commented:
Looks great! Very detailed and well-documents, looks good to me.

lhlee-umass reviewed (approved):
Well-documented, thanks for creating the mock data

lrdiaz reviewed (approved):
Everything looks finalized! Thank you for adding a more detailed description!
